### PR TITLE
[documentation] Adding a new Kubernetes Resource

### PR DESCRIPTION
**Pull request by natalysheinin**
**What this PR does / why we need it**: 

Adds documentation on how to add a new kubernetes resource and its metrics. This will help future contributors by indicating all the file additions/changes required to ship a PR that will pass tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -290,3 +291,15 @@ Then curl the metrics endpoint
...
+
+#### Add New Kubernetes Resource
+
+Adding a new Kubernetes resource and its respective metrics requires modifying a few more files than you may think.
+
+- Reference your new resource to the [docs/README.md](https://github.com/kubernetes/kube-state-metrics/blob/master/docs/README.md#exposed-metrics).
+- Reference your new resource in the [docs/cli-arguments.md](https://github.com/kubernetes/kube-state-metrics/blob/master/docs/cli-arguments.md#available-options) as part of the `--collectors` flag.
+- Create a new `<name-of-resource>.md` in the [docs](https://github.com/kubernetes/kube-state-metrics/tree/master/docs) directory to provide documentation on the resource(s) and metrics you implemented. Follow the formatting of all other resources.
+- Add the resource you are representing to the [examples/standard/cluster-role.yaml](https://github.com/kubernetes/kube-state-metrics/blob/master/examples/standard/cluster-role.yaml) under the appropriate `apiGroup` using the `verbs`: `list` and `watch`.
+- Reference the new resource in [pkg/options/collector.go](https://github.com/kubernetes/kube-state-metrics/blob/master/pkg/options/collector.go).
```
Comment (tariq1890) [inline README.md]: Let's add this link as well to the bullet point. 

https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/builder.go

**Comment by tariq1890** [comment]
@natalysheinin Thank you so much. This is MVP level!

**Comment by tariq1890** [comment]
I am still trying to think of the best place for this. Should it be in `README.md` or `CONTRIBUTING.md`

**Comment by tariq1890** [comment]
Now that I think more about this, we should have a developers guide doc.

File: README.md [hunk 2]
```diff
@@ -290,3 +291,15 @@ Then curl the metrics endpoint
 	curl localhost:8080/metrics
 
 To run the e2e tests locally see the documentation in [tests/README.md](./tests/README.md).
+
+#### Add New Kubernetes Resource
+
+Adding a new Kubernetes resource and its respective metrics requires modifying a few more files than you may think.
```
Comment (LiliC) [inline README.md]: I find this style of writing more fitting for a blog post. Lets just say something like:

```
The following steps are needed to introduce a new resource and respective resource metrics.
```

File: README.md [hunk 3]
```diff
@@ -290,3 +291,15 @@ Then curl the metrics endpoint
 	curl localhost:8080/metrics
 
 To run the e2e tests locally see the documentation in [tests/README.md](./tests/README.md).
+
+#### Add New Kubernetes Resource
```
Comment (LiliC) [inline README.md]: WDYT:

```suggestion
#### Add New Kubernetes Resource Metric Collector
```

File: README.md (hunk 2 above)
Comment (natalysheinin) [reply README.md]: SGTM

**Comment by natalysheinin** [comment]
@LiliC @tariq1890 which doc is the developers guide? or would that be a new one?

**Comment by LiliC** [comment]
@natalysheinin Yes one doesn't exist yet, if you don't mind creating it in the `docs/developer/guide.md` something along those lines and link it from the main README.md.

File: README.md [hunk 4]
```diff
@@ -290,3 +291,16 @@ Then curl the metrics endpoint
 	curl localhost:8080/metrics
 
 To run the e2e tests locally see the documentation in [tests/README.md](./tests/README.md).
+
+#### Add New Kubernetes Resource Metric Collector
+
+The following steps are needed to introduce a new resource and respective resource metrics.
```
Comment (tariq1890) [inline README.md]:
```suggestion
The following steps are needed to introduce a new resource and its respective resource metrics.
```

**Comment by natalysheinin** [comment]
@LiliC good to go?

**Pull request merged**